### ISSUE TITLE
PCT_RIBOSOMAL_BASES can sometimes be too high

### ISSUE THREAD

**Issue by alecw** [broadinstitute/picard#1091]
https://github.com/broadinstitute/picard/blob/b63d73e7d8f4c372a9aef201793b12074d0c2322/src/main/java/picard/analysis/directed/RnaSeqMetricsCollector.java#L171

PCT_RIBOSOMAL_BASES is computed by RIBOSOMAL_BASES/PF_ALIGNED_BASES.  However, in this code, RIBOSOMAL_BASES can be incremented by more than PF_ALIGNED_BASES, resulting in PCT_RIBOSOMAL_BASES that is too large.  I'm running into a problem downstream because the percentages don't always add up properly.  

I think the safe thing to do would be to add an additional metric, RIBOSOMAL_ALIGNED_BASES, tally the ribo bases both ways, but compute PCT_RIBOSOMAL_BASES as RIBOSOMAL_ALIGNED_BASES/PF_ALIGNED_BASES.  

Note that this bug has been around for ~7 years.  I'm not naming the responsible party, but suffice it to say that he knows who he is ;-)

I'm going to hack around this problem in my downstream code for now.  If folks are interested, I can submit a pull request.  Or we could just shrug and move on.

**Comment by yfarjoun** [comment]
This was due to a code change that was introduced in 2011 where the number
of RIBOSOMAL bases added to the tally is the read-length. I'm not sure that
the correct number is the number of bases in the read...it seems more
reasonable that number of aligned bases in the read are a better
approximation of the number of ribosomal bases than the length of the read.
I suspect that with the short read-data of 2011 this didn't matter, but
with longer reads this can cause (as you see) a real difference.

In short, I would vote for changing the RIBOSOMAL_BASES logic to `
RIBOSOMAL_BASES+=getNumAlignedBases(rec)`.

Keeping around both RIBOSOMAL_ALIGNED_BASES and RIBOSOMAL_BASES doesn't
make much sense to me.

**Comment by alecw** [comment]
> In short, I would vote for changing the RIBOSOMAL_BASES logic to `
RIBOSOMAL_BASES+=getNumAlignedBases(rec)`.

That would be fine with me.  The motivation for my more complicated proposal was to change as little as possible of existing results, but I think what you suggest is cleaner.
